homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,230 @@
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        from aioswitcher.consts import STATE_ON as SWITCHER_STATE_ON
+        return cast(bool, self._state == SWITCHER_STATE_ON)
+
+    @property
+    def current_power_w(self) -> int:
+        """Return the current power usage in W."""
+        return cast(int, self._device_data.power_consumption)
+
+    @property
+    def device_state_attributes(self) -> Dict:
+        """Return the optional state attributes."""
+        from aioswitcher.consts import WAITING_TEXT
+
+        attribs = {}
+
+        for prop, attr in PROPERTIES_TO_ATTRIBUTES.items():
+            value = getattr(self, prop)
+            if value and value is not WAITING_TEXT:
+                attribs[attr] = value
+
+        return attribs
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        from aioswitcher.consts import (STATE_OFF as SWITCHER_STATE_OFF,
+                                        STATE_ON as SWITCHER_STATE_ON)
+        return self._state in [SWITCHER_STATE_ON, SWITCHER_STATE_OFF]
+
+    async def async_update_data(self, event: Event) -> None:
+        """Update the entity data."""
+        device_data = event.data.get(UPDATED_DEVICE)
+        if device_data:
+            if self._self_initiated:
+                self._self_initiated = False
+            else:
+                self._device_data = device_data
+                self._state = self._device_data.state
+                self.async_schedule_update_ha_state()
+
+        return None
+
+    async def async_turn_on(self, **kwargs: Dict) -> None:
+        """Turn the entity on.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        from aioswitcher.api import SwitcherV2Api
+        # pylint: disable=unused-import
+        from aioswitcher.api.messages import (  # noqa F401
+            SwitcherV2ControlResponseMSG)
+        # pylint: enable=unused-import
+        from aioswitcher.consts import (COMMAND_ON,
+                                        STATE_ON as SWITCHER_STATE_ON)
+        response = None  # type: SwitcherV2ControlResponseMSG
+        async with SwitcherV2Api(
+                self._hass.loop, self.device_ip_addr,
+                self._device_data.phone_id, self._device_data.device_id,
+                self._device_data.device_password) as swapi:
+            response = await swapi.control_device(COMMAND_ON)
+
+        if response and response.successful:
+            self._self_initiated = True
+            self._state = SWITCHER_STATE_ON
+            self.async_schedule_update_ha_state()
+
+        return None
+
+    async def async_turn_off(self, **kwargs: Dict) -> None:
```

Comment:
These methods only really differ by a couple lines -- could you create a shared function instead to eliminate the repetitive code?

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/bcc8608dc5ce4fb4b48ed43a276dec0999ced242).